README.md

```diff
@@ -29,6 +30,206 @@ runApp(...
 
 2. Run the application in debug mode and watch for a leak related warnings. If you see a warning, open the link to investigate the leaks.
 
-TODO(polina-c): add example of the warning
+TODO(polina-c): implement the link and add example of the warning.
+
+### Flutter tests
+
+Wrap your tests with `withLeakTracking` to setup automated leak verification:
+
+```dart
+test('...', () async {
+  final leaks = await withLeakTracking(
+    () async {
+      ...
+    },
+    throwOnLeaks: false,
+  );
+
+  expect(leaks.total, 0);
+});
+```
+
+## Leak tracking concepts
+
+Before reading about leak tracking, understand [Dart memory concepts](https://docs.flutter.dev/development/tools/devtools/memory#basic-memory-concepts).
+
+### Addressed leak types
+
+The leak tracker can catch only certain types of leaks, in particular, related to timing of disposal and garbage collection.
+
+The tool assumes that, with proper memory management,
+an object's disposal and garbage collection should happen sequentially,
+close to each other.
+
+By monitoring disposal and GC events, the tool detects different types of leaks:
+
+**Not disposed, but GCed (not-disposed)**: a disposable object was GCed,
```

Comment:
It may be good to split the bodies of these enties into "Description" and "Fix" bullets, so that it is easy to visually parse the sections and their value.